### PR TITLE
add retry for getting node when setup

### PR DESCRIPTION
**Pull request by yuyulei**
when machine reboot, local-volume-provisioner started as daemonset, try to get node info with ClusterIP of svc:kubernetes. 

But at the same time, kube-proxy maybe not start, so the iptable rule(ClusterIP of svc:kubernetes -> apiserver address) has not been created/updated. 

So find this :

`F0515 11:48:50.664856       1 main.go:119] Could not get node information: Get https://<ClusterIP of  svc:kubernetes>:443/api/v1/nodes/<node_name>: dial tcp <ClusterIP>:443: connect: connection timed out`

After kube-proxy started, local-volume-provisioner will start successfully. 

Add retry can protect local-volume-provisioner from crashing.

### CODE REVIEW

**Comment by cofyc** [comment]
hi, `local-volume-provisioner` is a long running program, you should set `restartPolicy` to `Always`.  The kubelet will restart it if it fails to get the node. It's best practice to fail on fatal errors in production and let process supervisor restart the process.

**Comment by yuyulei** [comment]
In actual, this crash is stable probability to reproduce. 

Esay retries can help to avoid an expected crash. Now the provisioner is stateless, so it works by restarting. 

In the actual production environment, there maybe are some monitors watching all kinds failure of pods and nodes. Assuming that reboot happens, the monitor will report the crash of provisioner possibly, but we no need to handle this. The monitor just confuse us .

File: cmd/local-volume-provisioner/main.go [hunk 1]
```diff
@@ -111,9 +113,17 @@ func main() {
...
-		klog.Fatalf("Could not get node information: %v", err)
+	var retries int
+
+	for {
+		node, err := client.CoreV1().Nodes().Get(name, metav1.GetOptions{})
+		if err == nil {
+			return node
+		}
+
+		retries++
```
Comment (cofyc) [inline cmd/local-volume-provisioner/main.go]: add a log here and one-second sleep, then the user knows what happens, e.g.

```
klog.Infof("Could not get node information (remaining retries: %d): %v", maxGetNodesRetries - retries, err)
time.Sleep(time.Second)
```

File: cmd/local-volume-provisioner/main.go [hunk 2]
```diff
@@ -111,9 +113,17 @@ func main() {
...
+		if retries == MaxGetNodesRetries {
```
Comment (cofyc) [inline cmd/local-volume-provisioner/main.go]: `==` => `>=`

File: cmd/local-volume-provisioner/main.go [hunk 3]
```diff
@@ -38,6 +38,8 @@ import (
 	"k8s.io/client-go/kubernetes"
 )
 
+const MaxGetNodesRetries = 3
+
```
Comment (cofyc) [inline cmd/local-volume-provisioner/main.go]: `MaxGetNodesRetries` -> `maxGetNodesRetries`

File: cmd/local-volume-provisioner/main.go (hunk 1 above)
Comment (cofyc) [reply cmd/local-volume-provisioner/main.go]: a small sleep ensures that the provisioner will not fail in a short time if we want it to tolerate network failures.

**Comment by cofyc** [comment]
/lgtm

hi, @yuyulei Thanks! 

By the way, you need to sign the CLA. You can follow instructions [here](https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/pull/102#issuecomment-493316165).

**Comment by yuyulei** [comment]
I signed it

**Comment by cofyc** [comment]
/lgtm
/approve
Thanks for your contribution!

**Pull request merged**